guides/plugins/plugins/content/sitemap/modify-sitemap-entries.md

```diff
@@ -0,0 +1,165 @@
+# Modify sitemap entries
+
+## Overview
+
+You might have had a look at our guide about [adding custom sitemap entries](add-custom-sitemap-entries.md),
+e.g. for a custom entity.
+However, you might not want to add new URLs, but rather modify already existing ones.
+This guide will cover modifying e.g. the product URLs for the sitemap.
+
+## Prerequisites
+
+This guide is built upon the [plugin base guide](../../plugin-base-guide.md), like most guides.
+
+Modifying the sitemap entries is done via decoration, so should know how that's done as well.
+Also, knowing how the URL providers work, like it's explained in our guide about [adding custom sitemap entries](add-custom-sitemap-entries.md),
+will come in handy.
+
+## Modifying the sitemap
+
+There's two ways of actually modifying the sitemap entries, but both ways are done by decorating
+the respective `UrlProvider`, e.g. the `Shopware\Core\Content\Sitemap\Provider\ProductUrlProvider` for products.
+
+Hence, let's start with creating the basic decorated class for the `ProductUrlProvider`. We'll call
+this class `DecoratedProductUrlProvider`:
+
+{% tabs %}
+{% tab title="DecoratedProductUrlProvider.php" %}
+{% code title="<plugin root>/src/Core/Content/Sitemap/Provider/DecoratedProductUrlProvider.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Core\Content\Sitemap\Provider;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Content\Sitemap\Provider\AbstractUrlProvider;
+use Shopware\Core\Content\Sitemap\Struct\UrlResult;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+
+class DecoratedProductUrlProvider extends AbstractUrlProvider
+{
+    /**
+     * @var AbstractUrlProvider
+     */
+    private $decoratedUrlProvider;
+
+    public function __construct(AbstractUrlProvider $abstractUrlProvider)
+    {
+        $this->decoratedUrlProvider = $abstractUrlProvider;
+    }
+
+    public function getDecorated(): AbstractUrlProvider
+    {
+        return $this->decoratedUrlProvider;
+    }
+
+    public function getName(): string
+    {
+        return $this->getDecorated()->getName();
+    }
+
+    public function getUrls(SalesChannelContext $context, int $limit, ?int $offset = null): UrlResult
+    {
+        return $this->getDecorated()->getUrls($context, $limit, $offset);
+    }
+}
+```
+{% endcode %}
+{% endtab %}
+
+{% tab title="services.xml" %}
+{% code title="<plugin root>/src/Resources/config/services.xml" %}
+```markup
```

Comment:
I guess you want to highlight xml, right?

Reply:
Wait, I'm pretty sure I copied & pasted it. There might be more wrong occurrences, thanks! Will fix them all in an extra commit.

Reply:
Welp... The GitBook linter changed a lot of them to `markup` (including twig, html and xml...)

Reply:
still reads weird. Probably other markdown renderers are more verbose